presto-main/src/main/java/io/prestosql/sql/gen/LambdaBytecodeGenerator.java

```diff
@@ -142,6 +142,14 @@ public static CompiledLambda preGenerateLambdaExpression(
                 lambdaExpression);
     }
 
+    /**
+     * Parameter name for the generated method.
+     */
+    private static String parameterName(int argumentPosition, String lambdaArgumentName)
```

Comment:
> Another place to do this sanitization could be SymbolAllocator#newSymbol. 

I thought about that. IMO the change here does not preclude change in `SymbolAllocator`.
And the change in `SymbolAllocator` does not make the change here obsolete:
compiler code should make as few assumptions about planner as possible.
It would feel wrong if planner had to consider Java class format constraints. 

For  #5088, as you pointed out, there are other benefits from keeping symbols short.
Plan readability, but also guard against any potential perf penalties
(at least in theory, a potential DOS attack vector).